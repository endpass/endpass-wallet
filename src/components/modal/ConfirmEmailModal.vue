<template>
  <v-modal
    class="is-dark"
    @close="handleClose"
  >
    <template slot="header">Check Your Email</template>

    <p class="subtitle">We just sent you an email with a confirmation link.
    Click it to instantly log in to your wallet!</p>
  </v-modal>
</template>

<script>
export default {
  name: 'ConfirmEmailModal',
  methods: {
    handleClose() {
      this.$emit('close');
    },
  },
};
</script>
