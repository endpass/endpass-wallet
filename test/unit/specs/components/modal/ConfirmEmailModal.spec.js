import { shallow } from '@vue/test-utils';

import ConfirmEmailModal from '@/components/modal/ConfirmEmailModal';

describe('ConfirmEmailModal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(ConfirmEmailModal);
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('ConfirmEmailModal');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('handleClose', () => {
      it('should trigger "close" event', () => {
        wrapper.vm.handleClose();

        expect(wrapper.emitted().close).toEqual([[]]);
      });
    });
  });
});
