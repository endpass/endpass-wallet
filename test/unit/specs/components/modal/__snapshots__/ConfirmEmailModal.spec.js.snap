// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailModal render should render initial state of the component 1`] = `
<div
  class="modal is-active is-dark"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-logo"
  />
   
  <div
    class="modal-card"
  >
    <div
      class="modal-card-head"
    >
      <h3
        class="modal-card-title"
      >
        Check Your Email
      </h3>
       
      <button
        class="delete is-large"
      />
    </div>
     
    <div
      class="modal-card-body"
    >
       
      <p
        class="subtitle"
      >
        We just sent you an email with a confirmation link.
  Click it to instantly log in to your wallet!
      </p>
    </div>
     
    <div
      class="modal-card-foot"
    />
  </div>
</div>
`;
